### ISSUE TITLE
Make PVI detail DeviceVectors

### ISSUE THREAD

**Issue by coretl** [bluesky/ophyd-async#624]
Could: Is this there because PVI has a standard that says something like "`thing_n` is part of a list of 'thing's on the device"? Or is it more of a loose convention?

Either way, since we get to decide what PVI says, could we make it return a structure rather than a string to make it clear this is the case? Out of scope for this PR obviously, would be a separate issue.

_Originally posted by @callumforrester in https://github.com/bluesky/ophyd-async/pull/594#discussion_r1812340170_
            

Can do this by using QSRV structures with `[n]` in them:
```
    info(Q:group, {
        "$(IOC_NAME=PANDAQSRV):PVI": {
            "value.pulse[1].d": {
                "+channel": "VAL",
                "+type": "plain"
            }
        }
    })
```
We can make this back compatible with https://github.com/PandABlocks/PandABlocks-ioc/issues/138

**Comment by stan-dot** [comment]
@coretl not clear what is the use case exactly here. is it just syntactic sugar?

**Comment by coretl** [comment]
Fixed by #625
